tests/testdata/default_config.yaml

```diff
@@ -208,6 +208,24 @@ generate:
       #   - ['--dtype', 'auto']
       #   - ['--lora-alpha', '32']
       vllm_args: []
+# Ingest configuration section.
+ingest:
+  # Document store configuration for RAG.
+  # Default: uri='embeddings.db' collection_name='ilab'
+  document_store:
+    # Document store collection name.
+    # Default: ilab
+    collection_name: ilab
+    # Document store service URI.
+    # Default: embeddings.db
+    uri: embeddings.db
+  # Embedding parameters for ingestion.
+  # Default: embedding_model_name='/Users/dmartino/.cache/instructlab/models/ibm-granite/granite-embedding-125m-english'
```

Comment:
@cdoern can you explain how to generate the default_config.yaml without references to the local dev environment? I used `make regenerate-test-data` to update it, but this is not acceptable when it runs on the GH CI environment. 
[failing check](https://github.com/instructlab/instructlab/actions/runs/12752699917/job/35542994974?pr=2903#step:15:212)

Reply:
if you `make regenerate-testdata` and then commit the generated `default_config.yaml` that should work if it passed locally. I will look at the failing tests

Reply:
Got it: we used the constructor to initialize the default and the below mock replacement did not work. Fixed by using `default_factory` instead. Thanks!

```py
@patch.multiple(
    config.DEFAULTS,
    _cache_home="/cache/instructlab",
    _data_dir="/data/instructlab",
)
def test_compare_default_config_testdata(
    testdata_path: pathlib.Path, regenerate_testdata: bool
):
...
```